### PR TITLE
Add GitHub actions

### PR DESCRIPTION
**Pull request by somexpert**
This adds support for github actions, removes Travis CI, switches to pytest as our test runner, and adds support for Python 3.8 - 3.11

### CODE REVIEW

**Comment by somexpert** [comment]
ping @ldko, @gracieflores, this is ready for review

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 Change Log
 ==========
 
+4.1.0
+-----
+* Added support for Python 3.8 - 3.11
+* Dropped Travis CI integration
+* Moved CI to Github Actions
+* Switched to using pytest to run test suite
```
Comment (gracieflores) [inline CHANGELOG.md:9]: I would add a period at the end of each of these, just for consistency.

**Review by gracieflores — COMMENTED** [review]
Other than my one comment, I think this looks good. 👍🏼

File: CHANGELOG.md (hunk 1 above)
Comment (somexpert) [reply CHANGELOG.md:9]: oh, yes, I'll add those!

**Review by ldko — APPROVED** [review]
Works for me :+1: . Thanks, @somexpert !

**Pull request merged**